src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/taxon.yml

```diff
@@ -50,7 +50,7 @@ sylius_admin_ajax_taxon_by_name_phrase:
                 method: findByNamePart
                 arguments:
                     phrase: $phrase
-                    locale: null
+                    locale: expr:service('sylius.context.locale').getLocaleCode()
```

Comment:
I'm not sure about this change. In my opinion, this causes a problem with autocompletes, but I don't know why it was set to null?

Reply:
When it's null, it searches though the translations in all locales, when a locale is passed, it will only search though the translations in that locale.
https://github.com/Sylius/Sylius/blob/f42d92602683a41dabe4fd5514f6882aa4e48908/src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php#L93-L102
https://github.com/Sylius/Sylius/blob/f42d92602683a41dabe4fd5514f6882aa4e48908/src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php#L109-L124

Reply:
In other words, this will limit the search to the admin's locale, which will create another issue.
The way I would approach it, is by using the translation which matched the query to display the name / fullname.

Reply:
Hmm, I did not specify my question, sorry 😞 I should ask why limiting the search to the admin's locale will create another issue? Now we are searching for all translations for eg. `LIKE "%t%"` and if found it hydrates it to your locale eg. `en_US` and this causes a problem because we found two additional categories which are not in your locale and it is not translating to `en_US` because we don't have this translation (that's why the name was undefined) and I think limiting to admin’s locale is ok. But if I’m wrong please correct me 😃

Reply:
The question is: why limit it to the admin's locale when you can search though all translations?
It's an unnecessary limitation and I'm sure users are going to complain about it just like they did regarding other parts of the admin panel: https://github.com/Sylius/Sylius/issues/13296, https://github.com/Sylius/Sylius/issues/13134, https://github.com/Sylius/Sylius/issues/12145
The admin locale should be used mostly for translating the UI, not to restrict the content.

Reply:
To emphasize it even more, consider that the admin can choose **ANY** locale and if it's not one of the enabled locale in the store, the the admin won't see no results.

Reply:
To help you, here are two ideas of how this could be solved:
1. Add `Autocomplete` group, to Taxon's `translations` property and to the relevant properties in TaxonTranslation, in their serialization config. Enabled the `sylius-auto-complete.js` to access nested attributes, so you could pass a path like `translations[0].fullname` to `choice_name`  option of `TaxonAutocompleteChoiceType`.
2. Loop though the result set and set the fallback locale for each taxon to the locale of it's first translation. This can be done either directly in the repository method, or using the index event.

Reply:
Ok, I think I understand this, thanks 😃